src/state/nfts/types.ts

```diff
@@ -0,0 +1,69 @@
+import { Address } from 'viem';
+import { QueryStoreState } from '@/state/internal/queryStore/types';
+import { OptionallyPersistedRainbowStore } from '@/state/internal/types';
+import { UniqueAsset } from '@/entities';
+
+export type CollectionId = string;
+export type UniqueId = string;
+export type CollectionName = string | 'Showcase';
+
+export type NftParams = {
+  walletAddress: Address | string;
+  limit: number;
+  pageKey: string | null;
+  openCollections?: CollectionId[];
+};
+
+export type PaginationInfo = {
+  pageKey: string | null;
+  hasNext: boolean;
+  total_elements?: number;
+};
+
+export type NftsQueryData = {
+  collections: Map<CollectionId, Collection>;
+  nftsByCollection: Map<CollectionId, Map<UniqueId, UniqueAsset>>;
+  pagination: PaginationInfo | null;
+};
+
+export type Collection = {
+  id: CollectionId;
+  name: string;
+  imageUrl: string | null | undefined;
+  totalCount: string;
+};
+
+export interface NftsState {
+  collections: Map<CollectionId, Collection>;
+  nftsByCollection: Map<CollectionId, Map<UniqueId, UniqueAsset>>;
+  openCollections: Set<CollectionId>;
+  fetchedPages: { [pageKey: string]: number };
+  fetchedCollections: { [collectionId: string]: number };
+  pagination: PaginationInfo | null;
+  fetchNextPage: () => Promise<void>;
+  getCollections: () => Collection[] | null;
```

Comment:
one thing i'm trying to do with stores in general is keep the object name in there so like `getNFTCollections`, that way you can export them statically and import them elsewhere and its not polluting other actions